### DISCUSSION TITLE
How to use no-$ref-siblings rule from oas3_0 in oas3_1?

### DISCUSSION THREAD

**Discussion by clydecutting** [stoplightio/spectral#2340] [Q&A]
How can I use no-$ref-siblings rule from oas3_0 in oas3_1?

Against an OAS 3.1 format file that rule appears to be undefined. These efforts have failed:

(TBD...)

**Comment by cuttingclyde** [comment]
Here are the failed efforts:

1:
`extends: [[spectral:oas, all]]`
`rules:`
`  no-$ref-siblings: warn`

and 
2: `no-$ref-siblings: on`
and `no-$ref-siblings: "on" `

Give spectral error `Error at #/rules/no-$ref-siblings: must be object`.

3:
`extends: [[spectral:oas, all]]`
`rules:`
`  no-$ref-siblings:`
`    description: 'Property must not be placed among $ref'`
`    message: '{{error}}'`
`    severity: warn`
`    resolved: false`
`    given: "$..[?(@property === '$ref')]"`
`    then:`
`      function: refSiblings`

Gives spectral error `Error at #/rules/no-$ref-siblings/then: must have required property 'function'`
(Which I see is to be expected since refSiblings is not a standard spectral function. Could be?)

4:
`extends: [[spectral:oas, all]]`
`overrides:`
`  - formats:`
`      - oas3_1`
`    rules:`
`      no-$ref-siblings: "on" `

Gives error: `Cannot read property 'replace' of undefined`
